[refactor] Move some of the helper functions for public use

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #32379 [WIP][quant][graphmode] FoldConvBatchNorm2d support shared ClassTypes
* #32380 [quant][graphmode][refactor] Better API for fold_convbn
* #32374 [refactor] Adding FoldConvBatchNorm2dHelper
* #32303 [quant][graphmode] Call _jit_pass_dedup_module_ueses in quantize_script
* #32302 [quant] Re-enable fold_convbn in quantize_script
* #32206 [quant] Re-enable test_nested that has different qconfig for shared ClassType
* #32205 [quant][graphmode] Support quantizing shared ClassType with different qconfigs
* #32204 [quant][graphmode] Default to non-inplace in graph mode quantization API
* #32203 Adding QConfigTypePtrMap
* **#32202 [refactor] Move some of the helper functions for public use**
* #31408 [quant][graphmode] `insert_quant_dequant` pass support shared class types

Summary:
Move some helper functions in ModuleUseDeduper for public use

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D19508034](https://our.internmc.facebook.com/intern/diff/D19508034)

jerryzh168 committed:
[refactor] Move some of the helper functions for public use

Summary:
Move some helper functions in ModuleUseDeduper for public use

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 9047840c:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=9047840c2762379dbf4701e5e12c0ba12c6f1d1f).

### :detective: 1 new failure recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4248130) pytorch_xla_linux_xenial_py3_6_clang7_build (1/1)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=86222561) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=86222561))
<details>
<summary>
<code>Jan 15 02:38:56 RuntimeError: Failed to apply patch: /var/lib/jenkins/workspace/xla/torch_patches/X10-clip_grad.diff</code>
</summary>

```
Jan 15 02:38:56 + XDIR=/var/lib/jenkins/workspace/xla/scripts/.. 
Jan 15 02:38:56 + PTDIR=/var/lib/jenkins/workspace/xla/scripts/../.. 
Jan 15 02:38:56 + python /var/lib/jenkins/workspace/xla/scripts/cond_patch.py /var/lib/jenkins/workspace/xla/scripts/../torch_patches /var/lib/jenkins/workspace/xla/scripts/../.. 
Jan 15 02:38:56 1 out of 1 hunk FAILED 
Jan 15 02:38:56 Applying patch file: /var/lib/jenkins/workspace/xla/torch_patches/X10-clip_grad.diff 
Jan 15 02:38:56 Traceback (most recent call last): 
Jan 15 02:38:56   File "/var/lib/jenkins/workspace/xla/scripts/cond_patch.py", line 67, in <module> 
Jan 15 02:38:56     patch_repo(args) 
Jan 15 02:38:56   File "/var/lib/jenkins/workspace/xla/scripts/cond_patch.py", line 49, in patch_repo 
Jan 15 02:38:56     raise RuntimeError('Failed to apply patch: {}'.format(ppath)) 
Jan 15 02:38:56 RuntimeError: Failed to apply patch: /var/lib/jenkins/workspace/xla/torch_patches/X10-clip_grad.diff 
Jan 15 02:38:56 + cleanup 
Jan 15 02:38:56 + retcode=1 
Jan 15 02:38:56 + set +x 
Jan 15 02:38:56 =================== sccache compilation log =================== 
Jan 15 02:38:56 =========== If your build fails, please take a look at the log above for possible reasons =========== 
Jan 15 02:38:56 Compile requests                 1 
Jan 15 02:38:56 Compile requests executed        0 
Jan 15 02:38:56 Cache hits                       0 
Jan 15 02:38:56 Cache misses                     0 
Jan 15 02:38:56 Cache timeouts                   0 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

jerryzh168 committed:
Update on "[refactor] Move some of the helper functions for public use"


Summary:
Move some helper functions in ModuleUseDeduper for public use

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move some of the helper functions for public use"


Summary:
Move some helper functions in ModuleUseDeduper for public use

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move some of the helper functions for public use"


Summary:
Move some helper functions in ModuleUseDeduper for public use

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jamesr66a commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -148,6 +148,37 @@ bool valueNeedsToBeQuantized(Value* v) {
+  Graph* graph = value->owningGraph();
+  const auto& inputs = graph->inputs();
+  return std::find(inputs.begin(), inputs.end(), value) != inputs.end();
+}
```
This might be eligible to be an API in `ir.h`, e.g. `Value::isOwningGraphInput()`

jerryzh168 replied:
we can move this to Value is more use cases comes up :)

jamesr66a replied:
I think that makes sense, but another thing to consider is that someone might implement basically the same thing and they wouldn't know to talk to us or look in this specific file. So putting it there preemptively ensures new callsites use a common API. Not high-pri though

jerryzh168 replied:
Yeah, that's true. I'm just not sure how useful this function is, if there is one more use case then it justifies us putting this into `ir.h` I guess.

jerryzh168 committed:
Update on "[refactor] Move some of the helper functions for public use"


Summary:
Move some helper functions in ModuleUseDeduper for public use

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move some of the helper functions for public use"


Summary:
Move some helper functions in ModuleUseDeduper for public use

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move some of the helper functions for public use"


Summary:
Move some helper functions in ModuleUseDeduper for public use

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move some of the helper functions for public use"


Summary:
Move some helper functions in ModuleUseDeduper for public use

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move some of the helper functions for public use"


Summary:
Move some helper functions in ModuleUseDeduper for public use

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]
